importers/yelp.ca/generate_csv.py

```diff
@@ -0,0 +1,127 @@
+            # searches for keyword in reviews and returns list containing sentences containing keyword
+            accessible = re.findall(r"([^.]*?accessible[^.]*\.)",reviews)
+
+        business_info = get_text_from_css(root,'div.ywidget > ul')
+        business_info = business_info.replace(" ","").split('\n')
+        filtered = []
+        for i in business_info:
+            if i != "":
+                filtered.append(i)
+        dict_business_info = dict(itertools.izip_longest(*[iter(filtered)] * 2, fillvalue = "" ))
+        # looking for comments with information regarding accessibility incase wheelchair accessible is empty or absent        if dict_business_info['WheelchairAccessible'] == "Yes":
+        if 'WheelchairAccessible' in filtered:
+            wheelchair_accessible = dict_business_info['WheelchairAccessible']
+        elif wheelchair:
+            misc_info = wheelchair[0]
+        elif accessible:
+            misc_info = accessible[0]
+        else:
+            wheelchair_accessible = "N/A"
+            misc_info = "Call restaurant/store to verify accessibility."
+        if "Wi-Fi" in filtered:
+            wifi = dict_business_info['Wi-Fi']
+        else:
+            wifi = "N/A"
+        if "Parking" in filtered:
+            parking = dict_business_info['Parking']
+        else:
+            parking = "N/A"
+        return {
+            'name': biz_name,
+            'address': address,
+            'longitude': longitude,
+            'latitude': latitude,
+            'wheelchair accessible': wheelchair_accessible,
+            'wifi': wifi,
+            'phone number': phone_number,
+            'parking': parking,
+            'map': yelp_base_url + location_url,
+            'Misc Info': misc_info
+            }
+
+
+def get_all_downloaded_locations():
+    """
+    loops through directory and returns locations list with filenames
+    that begin with raw_html
+    """
+    locations = []
+    data_dir = 'data/'
+    for (dirname, dirs, files) in os.walk('.'):
+        for filename in files:
+            if filename.startswith('raw_html'):
+                filename = data_dir + filename
+                location_details = get_location_info(filename)
+                locations.append(location_details)
+    return locations
+
+
+def generate_csv():
+    print "Gathering info.."
+    locations = get_all_downloaded_locations()
+    keys = ['name','address','latitude','longitude','wheelchair accessible','wifi','phone number','parking', 'map', 'Misc Info']
+    for i in range(len(locations)):
+        filename = (locations[i]['name']).encode('utf-8') + '.csv'
+        print "Generating csv: " + filename
+        with open(filename, 'w') as csv_file:
+            csv_writer = csv.DictWriter(csv_file, fieldnames=keys)
+            csv_writer.writeheader()
+            for location in locations:
+                csv_writer.writerow((location).encode('utf-8'))
```

Comment:
This line is why you're getting that error about "encode" of dict.  location is a dict which has no encode method.  encode is a method of the string type.

Something like this should work better and would fix the problem where you generate numerous CSV files:
```
def generate_csv():
    print "Gathering info.."
    locations = get_all_downloaded_locations()
    keys = ['name','address','latitude','longitude','wheelchair accessible','wifi','phone number','parking', 'map', 'Misc Info']
    filename = 'locations.csv'
    with open(filename, 'w') as csv_file:
        csv_writer = csv.DictWriter(csv_file, fieldnames=keys)
        csv_writer.writeheader()
        for location in locations:
            csv_writer.writerow(location)
```